controller/linodemachine_controller_helpers.go

```diff
@@ -74,8 +80,24 @@ func (r *LinodeMachineReconciler) newCreateConfig(ctx context.Context, machineSc
 
 		return nil, err
 	}
-	createConfig.Metadata = &linodego.InstanceMetadataOptions{
-		UserData: b64.StdEncoding.EncodeToString(bootstrapData),
+
+	region, err := machineScope.LinodeClient.GetRegion(ctx, machineScope.LinodeMachine.Spec.Region)
+	if err != nil {
+		return nil, err
+	}
+	regionMetadataSupport := slices.Contains(region.Capabilities, "Metadata")
+	if machineScope.LinodeMachine.Spec.UseStackScriptBootstrap || !regionMetadataSupport {
+		logger.Info(fmt.Sprintf("using StackScripts for bootstrapping. useStackScriptBootstrap: %b, regionMetadataSupport: %b",
+			&machineScope.LinodeMachine.Spec.UseStackScriptBootstrap, &regionMetadataSupport))
+		createConfig.StackScriptID = capiStackScriptID
+		createConfig.StackScriptData = map[string]string{
+			"label":    machineScope.LinodeMachine.Name,
+			"userdata": b64.StdEncoding.EncodeToString(bootstrapData),
+		}
+	} else {
+		createConfig.Metadata = &linodego.InstanceMetadataOptions{
+			UserData: b64.StdEncoding.EncodeToString(bootstrapData),
+		}
```

Comment:
nit: This structure biases the reader towards StackScript configurations. I think this logic should be reversed. We should try defaulting to using the (idiomatic™️) Metadata service first and then falling back on StackScripts if Metadata is not available.